packages/forms/frontend/sirius-components-widget-reference/src/modals/TransferModal.tsx

```diff
@@ -0,0 +1,251 @@
+    event.preventDefault();
+    if (state.draggingRightItemId) {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          right: prevState.right.filter((entry) => prevState.draggingRightItemId !== entry.id),
+          leftSelection: [],
+        };
+      });
+    }
+  };
+
+  const handleDragStart = (id: string) => {
+    setState((prevState) => {
+      return {
+        ...prevState,
+        draggingRightItemId: id,
+      };
+    });
+  };
+
+  const handleDragEnd = () => {
+    setState((prevState) => {
+      return {
+        ...prevState,
+        draggingRightItemId: undefined,
+      };
+    });
+  };
+
+  const handleDropRight = (event: React.DragEvent) => {
+    event.preventDefault();
+    const dragSourcesStringified = event.dataTransfer.getData(DRAG_SOURCES_TYPE);
+    if (dragSourcesStringified) {
+      const sources = JSON.parse(dragSourcesStringified);
+      if (Array.isArray(sources) && sources.length > 0) {
+        const entriesDragged = sources as SelectionEntry[];
+        setState((prevState) => {
+          return {
+            ...prevState,
+            right: prevState.right.concat(
+              entriesDragged.filter(
+                (newEntry) => !prevState.right.some((existingEntry) => existingEntry.id === newEntry.id)
+              )
+            ),
+            leftSelection: [],
+          };
+        });
+      }
+    }
+  };
+
+  const onClick = (event: React.MouseEvent<Element, MouseEvent>, item: SelectionEntry) => {
+    if (event.ctrlKey || event.metaKey) {
+      event.stopPropagation();
+      const isItemInSelection = state.rightSelection.find((entry) => entry.id === item.id);
+      const newSelection: SelectionEntry[] = isItemInSelection
+        ? state.rightSelection.filter((entry) => entry.id !== item.id)
+        : [...state.rightSelection, item];
+      setState((prevState) => {
+        return {
+          ...prevState,
+          rightSelection: newSelection,
+          leftSelection: newSelection,
+        };
+      });
+    } else {
+      setState((prevState) => {
+        return {
+          ...prevState,
+          rightSelection: [item],
+          leftSelection: [item],
+        };
+      });
+    }
+  };
+
+  const handleDispatchRight = () => {
+    setState((prevState) => {
+      return {
+        ...prevState,
+        right: prevState.right.concat(
+          prevState.leftSelection.filter(
+            (newEntry) => !prevState.right.some((existingEntry) => existingEntry.id === newEntry.id)
+          )
+        ),
+        leftSelection: [],
+      };
+    });
+  };
+
+  const handleDispatchLeft = () => {
+    setState((prevState) => {
+      return {
+        ...prevState,
+        right: prevState.right.filter((entry) => prevState.rightSelection.find((selected) => selected.id !== entry.id)),
+        rightSelection: [],
+        leftSelection: [],
+      };
+    });
+  };
+
+  return (
+    <Dialog
```

Comment:
I get an horizontal scrollbar on the `MuiDialogContent-root` under Chrome (not under Firefox):

![screenshot-2023-08-17_11-28](https://github.com/eclipse-sirius/sirius-web/assets/10608/0b4a8b5d-7317-4091-9c76-7077790780b6)

Reply:
Done

Reply:
Followup: I think the "Select" button should be renamed something else for the same reason.
"Apply" seems like a good choice as it is generic and reflects the fact the the actual model change is only effective when the user closes the modal.

Reply:
Done